mm2src/mm2_libp2p/src/atomicdex_behaviour.rs

```diff
@@ -286,6 +293,36 @@ impl AtomicDexBehaviour {
                 let future = request_peers(relays, req, self.request_response.sender(), response_tx);
                 self.spawn(future);
             },
+            AdexBehaviourCmd::RequestAddresses {
+                req,
+                peers_addresses,
+                response_tx,
+            } => {
+                let peers = peers_addresses
+                    .into_iter()
+                    .filter_map(|(peer, address)| match peer.parse() {
+                        Ok(p) => {
+                            let multi_address = match Multiaddr::from_str(&address) {
+                                Ok(addr) => addr,
+                                Err(e) => {
+                                    error!("Error on parse address {}: {:?}", address, e);
+                                    return None;
+                                },
+                            };
+                            let mut addresses = HashSet::new();
```

Comment:
To avoid a bit of overhead of creating the HashSet on every `RequestAddresses` we can add the address once in the `add_node_to_version_stat` call and then pass only peer ids here.

Reply:
I implemented this in the latest commit but just found a small issue with this approach. When the stats collector node restarts for any reason all the nodes' addresses will have to be readded to the swarm before using `start_version_stat_collection`. Maybe a kickstart function can help with this or adding another RPC to get the addresses from the database and adds them to the swarm, what do you think is the best approach for this?

Reply:
I think that kickstarting by reading addresses from DB and adding them in the `stat_collection_loop` before starting the actual loop should work. Additional RPC is an undesirable approach.